src/ResourceManagement/ResourceManager/Core/ResourceUtils.cs

```diff
@@ -144,5 +144,38 @@ public static AzureEnvironment ExtractAzureEnvironment(RestClient restClient)
             return environment;
         }
 
+        public static string ResourceTypeForApiVersion(string id)
+        {
+            string type = ResourceTypeFromResourceId(id);
+            string parent = ParentResourcePathFromResourceId(id);
+            while (parent != null)
```

Comment:
Using String.IsNullOrEmpty for null check of string value.

Reply:
CoreUtils class